be/src/vec/exec/join/vhash_join_node.cpp

```diff
@@ -77,18 +81,44 @@ struct ProcessHashTableBuild {
             inserted_rows.reserve(_batch_size);
         }
 
+        std::vector<size_t> cached_hash_values(CACHE_NUM);
+        std::vector<KeyHolderType> cached_key_holders(CACHE_NUM);
+
+        size_t cached_count = 0;
         for (size_t k = 0; k < _rows; ++k) {
+            if (UNLIKELY(k % CACHE_NUM == 0)) {
+                cached_count = std::min(CACHE_NUM, _rows - k);
+                for (size_t i = k; i < _rows && i < (k + CACHE_NUM); ++i) {
```

Comment:
for (size_t i = 0; i < cache_count; ++i) {
                    if constexpr (ignore_null) {
                if ((*null_map)[i+k]) {
                        if ((*null_map)[i+k]) {
                            continue;
                        }
                    }

                    cached_key_holders[i ] =
                            key_getter.get_key_holder(i + k, _join_node->_arena);
                    cached_hash_values[i ] = key_getter.get_hash(
                            hash_table_ctx.hash_table, cached_key_holders[i ]);
                }